client/jetpack-cloud/sections/partner-portal/credit-card-fields/credit-card-submit-button.tsx

```diff
@@ -54,30 +53,53 @@ export default function CreditCardSubmitButton( {
 		return __( 'Please wait…' );
 	}, [ formStatus, activeButtonText, __ ] );
 
+	const { setCardDataError, setFieldValue, setFieldError } = useDispatch( creditCardStore );
+
+	const handleButtonClick = () => {
+		debug( 'validating credit card fields' );
```

Comment:
We're also inlining the credit card validation function here, that way we can take advantage of the existing store values without having to pass the entire store object. We could keep the function separate, but then we'd still need to pass all the various state data and mapped action creators, which is an extra step of unnecessary complication IMHO.